controllers/update_test.go

```diff
@@ -1413,3 +1435,77 @@ func mergeBranchIntoHead(repo *git.Repository, pushBranch string) {
 	})
 	Expect(err).NotTo(HaveOccurred())
 }
+
+// copied from source-controller/pkg/git/libgit2/checkout.go
+func commitFile(repo *git2go.Repository, path, content string, time time.Time) (*git2go.Oid, error) {
```

Comment:
I think we should rewrite the commit logic using libgit2 https://github.com/fluxcd/image-automation-controller/blob/544e4870393435ec07c078c69dc9749149bc60c4/controllers/imageupdateautomation_controller.go#L571

Reply:
I rewrote the gitInitRepo with libgit2, which was enough to remove the need for the adhoc fix.

Can we go ahead with the libgit2 to `1.3.0` on this PR and then any further rewrite/replacement of go-git we can deal on a separate PR.